src/charm.py

```diff
@@ -182,6 +214,36 @@ def _apply_k8s_resources(self, force_conflicts: bool = False) -> None:
             else:
                 raise GenericCharmRuntimeError("K8S resources creation failed") from error
 
+        for i in range(0, 3):
+            while True:
```

Comment:
issue(blocking): we should handle the retry with charm logic rather than a loop, for example put the charm in waiting until the validator is up and try in the next reconciliation, potentially defer the event if necessary.